photon-server/src/main/java/org/photonvision/common/logging/Logger.java

```diff
@@ -121,6 +129,54 @@ public static void addFileAppender(Path logFilePath) {
         currentAppenders.add(new FileLogAppender(logFilePath));
     }
 
+    public static void cleanLogs(Path folderToClean) {
+
+        LinkedList<File> logFileList =
+                new LinkedList<>(Arrays.asList(folderToClean.toFile().listFiles()));
+        HashMap<File, Date> logFileStartDateMap = new HashMap<>();
+
+        // Remove any files from the list for which we can't parse a start date from their name.
+        // Simultaneously populate our HashMap with Date objects repeseting the file-name
+        // indicated log start time.
+        logFileList.removeIf(
+                new Predicate<File>() {
+                    @Override
+                    public boolean test(File arg0) {
+                        try {
+                            logFileStartDateMap.put(
+                                    arg0, ConfigManager.getInstance().logFnameToTA(arg0.getName()));
+                            return false;
+                        } catch (ParseException e) {
+                            return true;
+                        }
+                    }
```

Comment:
Would it be possible to replace this with a lambda? We use lambdas everywhere else for stuff like sorting or filtering lists.

Reply:
Made it into a lambda - I think I did it right? Runs and passes test the same way it did previously.